build-tools-internal/src/main/groovy/elasticsearch.fips.gradle

```diff
@@ -23,9 +23,15 @@ if (BuildParams.inFipsJvm) {
     File fipsSecurity = new File(fipsResourcesDir, javaSecurityFilename)
     File fipsPolicy = new File(fipsResourcesDir, 'fips_java.policy')
     File fipsTrustStore = new File(fipsResourcesDir, 'cacerts.bcfks')
-    def bcFips = dependencies.create('org.bouncycastle:bc-fips:1.0.2')
-    def bcTlsFips = dependencies.create('org.bouncycastle:bctls-fips:1.0.9')
-
+    def bcFips = dependencies.create('org.bouncycastle:bc-fips:1.0.2.4')
+    def bcTlsFips = dependencies.create('org.bouncycastle:bctls-fips:1.0.17')
+    def manualDebug = false; //change this to manually debug bouncy castle in an IDE
+    if(manualDebug) {
+      bcFips = dependencies.create('org.bouncycastle:bc-fips-debug:1.0.2.4')
+      bcTlsFips = dependencies.create('org.bouncycastle:bctls-fips:1.0.17'){
+        exclude group: 'org.bouncycastle', module: 'bc-fips'  // to avoid jar hell
+      }
+    }
```

Comment:
For my own education, I understand that this bit adds the (newly updated) bc jars to test clusters' classpath when running with `-Dtests.fips.enabled=true`. Below, the test clusters are also configured with the `fips_java.security` properties file that defines the cryptography providers ordering.
So then I understand that (some) ITs will continue to work in a properly configured FIPS jvm (with updated jars). But what about the unit tests? Do we need to configure the CI workers?

Reply:
Unit tests run in the same JVM, so the line below will add the BC jars to the classpath:
```
test.setClasspath(test.getClasspath().plus(extraFipsJarsConfiguration))
```
then line further down 
```
   task.systemProperty('java.security.properties', String.format(Locale.ROOT, "=%s", fipsSecurity))
```
wires in the JCA config needed to use those jars. As far as I am aware we don't need to explicitly configure CI workers for FIPS, but we should be testing both the oracle and non-oracle variant. I can follow up to ensure we are testing both flavor of JVMs.

Reply:
🤦 I see now, thanks for explaining.
I agree there should not be any changes required for the CI workers.